Add task categories feature

admin960084 opened the pull request:
- Added category field to task format
- Updated example tasks with categories
- Documented category usage in README

Closes #1

SQLtattoo commented on tasks.txt:
```diff
@@ -1,5 +1,5 @@
-[LOW][Personal] Explore GitHub features
-[MEDIUM][Work] Review pull requests
\ No newline at end of file
+[HIGH][Work] Learn Git basics - Due: 2025-12-15
```
```suggestion
[HIGH][Work] Learn Git basics - Due: 2025-12-15
```

SQLtattoo committed:
Add task categories feature

- Added category field to task format
- Updated example tasks with categories
- Documented category usage in README

Closes #1

admin960084 reviewed (commented):
Looks good.
